modules/survey_accounts/php/addsurvey.class.inc

```diff
@@ -247,7 +247,7 @@ class AddSurvey extends \NDB_Form
         }
         $this->tpl_data['success'] = true;
 
-        if ($email) {
+        if ($email && ($_REQUEST['fire_away'] !== 'Create survey')) {
```

Comment:
This shouldn't be using the superglobal, we should be using the PSR7 request object to get variables.